corim/signedcorim.go

```diff
@@ -126,9 +126,30 @@ func (o *SignedCorim) FromCOSE(buf []byte) error {
 	return nil
 }
 
+func (o *SignedCorim) AddSigningCert(der []byte) error {
+	if der == nil {
+		return errors.New("nil signing cert")
+	}
+	o.message.Headers.Protected[cose.HeaderLabelX5Chain] = [][]byte{der}
```

Comment:
We shouldn't directly fiddle with the COSE internal structure here.

Instead, create two new attributes in the `SignedCorim` object: `signingCert` and `intermediateCerts` and set those from here.

Then, in `Sign`, populate the COSE headers as needed.

This allows us more flexibility in case we decide to change the internal arrangement from x5chain to x5chain+x5t, or x5bag, etc.